einks/rks.cc

```diff
@@ -0,0 +1,686 @@
+
+  // Determine the number of electrons in the system
+  // The molecule object is built into all wavefunctions
+  int charge = molecule_->molecular_charge();
+  int nelec = 0;
+  for (int i = 0; i < molecule_->natom(); ++i) {
+    nelec += (int)molecule_->Z(i);
+  }
+  nelec -= charge;
+  if (nelec % 2) {
+    throw PSIEXCEPTION("This is only an RHF code, but you gave it an odd "
+                       "number of electrons.  Try again!");
+  }
+  ndocc_ = nelec / 2;
+
+  outfile->Printf("    There are %d doubly occupied orbitals\n", ndocc_);
+  molecule_->print();
+  if (print_ > 1) {
+    basisset_->print_detail();
+    options_.print();
+  }
+
+  // nso_ = basisset_->nso();
+
+  // Nuclear repulsion without a field
+  e_nuc_ = molecule_->nuclear_repulsion_energy({0, 0, 0});
+  outfile->Printf("\n    Nuclear repulsion energy: %16.8f\n\n", e_nuc_);
+
+  // Make a MintsHelper object to help
+  auto mints = std::make_shared<MintsHelper>(basisset_);
+
+  // These don't need to be declared, because they belong to the class
+  auto S_mat = mints->so_overlap();
+
+  // Core Hamiltonian is Kinetic + Potential
+  auto H_mat = mints->so_kinetic();
+
+  this->nirrep_ = (int)S_mat->nirrep();
+
+  occ_per_irrep_.resize(nirrep_);
+
+  H_mat->add(mints->so_potential());
+
+  H_.set_name("Hamiltonian");
+  S_.set_name("Overlap");
+
+  for (int i = 0; i < S_mat->nirrep(); i++) {
+    irrep_sizes_.push_back(S_mat->rowdim(i));
+    auto S_block = einsums::Tensor<double, 2>(
+        molecule_->irrep_labels().at(i), S_mat->rowdim(i), S_mat->coldim(i));
+
+    for (int j = 0; j < S_mat->rowdim(i); j++) {
+      for (int k = 0; k < S_mat->coldim(i); k++) {
+        S_block(j, k) = S_mat->get(i, j, k);
+      }
+    }
+
+    S_.push_block(S_block);
+  }
+
+  for (int i = 0; i < S_mat->nirrep(); i++) {
+    auto H_block = einsums::Tensor<double, 2>(
+        molecule_->irrep_labels().at(i), H_mat->rowdim(i), H_mat->coldim(i));
+
+    for (int j = 0; j < H_mat->rowdim(i); j++) {
+      for (int k = 0; k < H_mat->coldim(i); k++) {
+        H_block(j, k) = H_mat->get(i, j, k);
+      }
+    }
+
+    H_.push_block(H_block);
+  }
+
+  if (print_ > 3) {
+    fprintln(*outfile->stream(), S_);
+    fprintln(*outfile->stream(), H_);
+    outfile->stream()->flush();
+  }
+
+  outfile->Printf("    Forming JK object\n\n");
+  // Construct a JK object that compute J and K SCF matrices very efficiently
+  jk_ = JK::build_JK(basisset_, mintshelper_->get_basisset("DF_BASIS_SCF"),
+                     options_, false, Process::environment.get_memory() * 0.8);
+  jk_->set_do_K(false);
+
+  // This is a very heavy compute object, lets give it 80% of our total memory
+  // jk_->set_memory(Process::environment.get_memory() * 0.8);
+  jk_->initialize();
+  jk_->print_header();
+
+  std::optional<std::map<std::string, double>> tweaks(std::in_place);
+  func_ = std::make_shared<SuperFunctional>();
+  //func_ = SuperFunctional::XC_build(options_.get_str("DFT_FUNCTIONAL"), true, tweaks);
```

Comment:
Could someone look at this? It's commented right now, but no matter what I do, I get link errors when running. All the errors are from libfunctional.